charts/bitwardenrs/values.yaml

```diff
@@ -24,6 +24,31 @@ bitwardenrs:
       enabled: false
       name: ""
       tokenKey: ""
+  # External database configuration.
+  # Requires bitwardenrs/server >= 1.17.0 or bitwardenrs/server-{mysql,postgres} images
+  # ref: https://github.com/dani-garcia/bitwarden_rs/wiki/Using-the-MySQL-Backend
+  #      https://github.com/dani-garcia/bitwarden_rs/wiki/Using-the-PostgreSQL-Backend
+  externalDatabase:
+    enabled: false
+    # Supported values: 'mysql', 'postgresql'.
+    type: ""
```

Comment:
It might be worth implementing this as a `database: "sqlite" | "mysql" | "postgresql"` and configuring details for the specific database backend in a subchart options format, i.e.
```yaml
bitwardenrs:
  database: "postgresql"

postgresql:
   postgresqlPassword: ... 
```
Let me know it's preferred to do things that way.

Reply:
@alebastr that makes sense to me, but would be a breaking change.

Reply:
@onedr0p -- how about we introduce the breaking change with the default value of the old version? Bump major version and include reference to the breaking change in the readme.

Reply:
Oh, sorry, seems like I missed the notification with @onedr0p's response. 
I don't think it's breaking if the database support exists only within this PR and the default is an old behavior (sqlite). IMO, it's the right moment to make decision on that without breaking changes.

That being said, I still need to learn how to use dependency charts before implementing this change and I didn't have time to do that :smile:

Reply:
> @onedr0p -- how about we introduce the breaking change with the default value of the old version? Bump major version and include reference to the breaking change in the readme.

this is fine with me as well

Reply:
My attempts to pull external database configuration from dependency chart config section ended in vain for a couple of reasons:
 - getting database host/port from postgresql or mariadb configuration requires uncomfortable amounts of internal knowledge about the structure of these charts
 - password value for bitwardenrs should be percent-encoded and therefore cannot be shared with other charts

Thus, the `bitwardenrs.externalDatabase` section remains as is unless someone else figures out a better way to do this.
It might be still worthy to declare these dependencies in the chart though.